apis/field_error.go

```diff
@@ -274,6 +361,9 @@ func merge(errs []*FieldError) []*FieldError {
 	// Sort the flattened map.
 	sort.Slice(newErrs, func(i, j int) bool {
 		if newErrs[i].Message == newErrs[j].Message {
+			if newErrs[i].Details == newErrs[j].Details {
+				return newErrs[i].Level < newErrs[j].Level
```

Comment:
This _happens_ to sort `Error` before `Warning` -- I don't know if we want to be more explicit about this, or leave it as a happy coincidence.

Reply:
We do this so that the ordering is deterministic, otherwise (I'm guessing map ordering) the lines can flip-flop.

Reply:
My concern was that adding an "Alert" level would sort "Alert", "Error", "Warning".

I'm also wondering if we want to sort by Level before Message, but I suspect that most of the time there will only be a few errors/warnings.